interfaces/sourcegen/sourcegen/csharp/_SourceGenerator.py

```diff
@@ -0,0 +1,309 @@
+        self._known_funcs[name] = func
+
+        return func
+
+
+    def _get_property_text(self, clazz: str, c_name: str, cs_name: str):
+        getter = self._known_funcs.get(clazz + "_" + c_name)
+
+        if (getter):
+            # here we have found a simple scalar property
+            prop_type = getter.ret_type
+        else:
+            # here we have found an array-like property (string, double[])
+            getter = self._known_funcs[clazz + "_get" + c_name.capitalize()]
+            # this assumes the last param in the function is a pointer type,
+            # from which we determine the appropriate C# type
+            prop_type = self._prop_type_map[getter.params[-1].p_type]
+
+        setter = self._known_funcs.get(clazz + "_set" + c_name.capitalize())
+
+        if prop_type in ['int', 'double']:
+            text = f'''
+                public {prop_type} {cs_name}
+                {{
+                    get => InteropUtil.CheckReturn(
+                        LibCantera.{getter.name}(_handle));'''
+
+            if(setter):
+                text += f'''
+                    set => InteropUtil.CheckReturn(
+                        LibCantera.{setter.name}(_handle, value));'''
+
+            text += '''
+                }
+            '''
+        elif (prop_type == 'string'):
+            p_type = getter.params[1].p_type
+
+            # for get-string type functions we need to look up the type of the second
+            # (index 1) param for a cast because sometimes it's an int and other times
+            # its a nuint (size_t)
+            text = f'''
+                public unsafe string {cs_name}
+                {{
+                    get => InteropUtil.GetString(40, (length, buffer) =>
+                        LibCantera.{getter.name}(_handle, ({p_type}) length, buffer));
+            '''
+
+            if(setter):
+                text += f'''
+                    set => InteropUtil.CheckReturn(
+                        LibCantera.{setter.name}(_handle, value));'''
+
+            text += '''
+                }
+            '''
+        else:
+            raise ValueError(f'Unable to scaffold properties of type {prop_type}!')
+
+        return(normalize_indent(text))
+
+
+    def generate_source(self, incl_file: os.DirEntry, funcs: list[Func]):
+        cs_funcs = [self._convert_func(f) for f in funcs]
+
+        functions_text = '\n\n'.join((self._get_function_text(f) for f in cs_funcs))
+
+        interop_text = normalize_indent(f'''
+            {normalize_indent(self._preamble)}
+
+            using System.Runtime.InteropServices;
+
+            namespace Cantera.Interop;
+
+            static partial class LibCantera
+            {{
+                {normalize_indent(functions_text)}
+            }}
+        ''')
+
+        with open(self._out_dir + 'Interop.LibCantera.'
+                + incl_file.name + '.g.cs', 'w') as f:
+            f.write(interop_text)
```

Comment:
`pathlib.Path.write_text()` is your friend here 😄

Reply:
awesome! yep, all path strings and `os.foo` replaced with `Path`s and `Path.foo`